hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/FreezeUnfreezeTokenPrecompileSuite.java

```diff
@@ -200,4 +221,288 @@ final HapiSpec isFrozenHappyPathWithAliasLocalCall() {
                         //                                                                .withIsFrozen(false)))))))
                         );
     }
+
+    @HapiTest
+    final HapiSpec freezeUnfreezeFungibleWithNegativeCases() {
+        final AtomicReference<TokenID> withoutKeyID = new AtomicReference<>();
+        final AtomicReference<AccountID> accountWithoutKeyID = new AtomicReference<>();
+        final AtomicReference<TokenID> vanillaTokenID = new AtomicReference<>();
+        final AtomicReference<AccountID> accountID = new AtomicReference<>();
+        final var NO_KEY_FREEZE_TXN = "NO_KEY_FREEZE_TXN";
+        final var NO_KEY_UNFREEZE_TXN = "NO_KEY_UNFREEZE_TXN";
+        final var ACCOUNT_HAS_NO_KEY_TXN = "ACCOUNT_HAS_NO_KEY_TXN";
+
+        return defaultHapiSpec("freezeUnfreezeFungibleWithNegativeCases")
+                .given(
+                        uploadInitCode(FREEZE_CONTRACT),
+                        contractCreate(FREEZE_CONTRACT),
+                        newKeyNamed(FREEZE_KEY).shape(THRESHOLD_KEY_SHAPE.signedWith(sigs(ON, FREEZE_CONTRACT))),
+                        newKeyNamed(MULTI_KEY),
+                        cryptoCreate(ACCOUNT).balance(100 * ONE_HBAR).exposingCreatedIdTo(accountID::set),
+                        cryptoCreate(ACCOUNT_WITHOUT_KEY).exposingCreatedIdTo(accountWithoutKeyID::set),
+                        cryptoCreate(TOKEN_TREASURY),
+                        cryptoCreate(ADMIN_KEY),
+                        tokenCreate(TOKEN_WITHOUT_KEY).exposingCreatedIdTo(id -> withoutKeyID.set(asToken(id))),
+                        tokenCreate(VANILLA_TOKEN)
+                                .tokenType(FUNGIBLE_COMMON)
+                                .treasury(TOKEN_TREASURY)
+                                .freezeKey(FREEZE_KEY)
+                                .adminKey(ADMIN_KEY)
+                                .initialSupply(1_000)
+                                .exposingCreatedIdTo(id -> vanillaTokenID.set(asToken(id))),
+                        newKeyNamed(THRESHOLD_KEY).shape(THRESHOLD_KEY_SHAPE.signedWith(sigs(ON, FREEZE_CONTRACT))),
+                        tokenUpdate(VANILLA_TOKEN).freezeKey(THRESHOLD_KEY).signedByPayerAnd(ADMIN_KEY),
+                        tokenAssociate(ACCOUNT, VANILLA_TOKEN),
+                        cryptoTransfer(moving(500, VANILLA_TOKEN).between(TOKEN_TREASURY, ACCOUNT)))
+                .when(withOpContext((spec, opLog) -> allRunFor(
+                        spec,
+                        contractCall(
```

Comment:
A few comments to indicate what test case each contractCall testing would be nice.

Reply:
Added.